File: src/types/grassland/creation-generation.ts
export type CreationGenerationKind =
  | 'video_adaptation'
  | 'asset_image'
  | 'scene_image'
  | 'article'
  | 'moments_copy'
  | 'comedy_script'
  | 'assistant_guide'
export type CreationGenerationMode = 'independent' | 'task'
export type CreationGenerationResolution = 'platform' | 'byok'

export interface CreationGenerationSummary {
  id: string
  kind: CreationGenerationKind
  mode: CreationGenerationMode
  provider: string
  model: string | null
  resultTitle: string
  createdAt: string
}

export interface CreationGenerationResultMediaItem {
  mediaId: string
  imageUrl: string | null
  available: boolean
}

export interface CreationGenerationDetail extends CreationGenerationSummary {
  contextSnapshotId: string | null
  aiRunId: string | null
  resolution: CreationGenerationResolution
  platformModelVersion: number | null
  upstreamRunId: string | null
  promptText: string
  inputSummary: Record<string, unknown>
  result: Record<string, unknown>
  resultMedia: CreationGenerationResultMediaItem[]
}

export interface CreationGenerationPage {
  items: CreationGenerationSummary[]
  nextBefore: string | null
}

/** 组织审计视图摘要（任务书 #44 登记）：带 ownerAccountId（审计需要「谁」）。 */
export interface OrgCreationGenerationSummary extends CreationGenerationSummary {
  ownerAccountId: string
}

export const CREATION_GENERATION_KIND_LABELS: Record<CreationGenerationKind, string> = {
  video_adaptation: '视频改编',
  asset_image: '素材图',
  scene_image: '分镜图',
  article: '文章正文',
  moments_copy: '朋友圈文案',
  comedy_script: '喜剧脚本',
  assistant_guide: '创作引导',
}

File: platform-java/services/intelligence-service/src/main/java/com/grassland/intelligence/creationlineage/CreationGeneration.java
package com.grassland.intelligence.creationlineage;

import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.UUID;

/** One immutable prompt-to-result lineage row. */
public record CreationGeneration(
        UUID id,
        String ownerAccountId,
        String organizationId,
        Kind kind,
        Mode mode,
        UUID contextSnapshotId,
        UUID aiRunId,
        Resolution resolution,
        String provider,
        String model,
        Integer platformModelVersion,
        String upstreamRunId,
        String promptText,
        Map<String, Object> inputSummary,
        List<UUID> inputMediaIds,
        Map<String, Object> result,
        List<UUID> resultMediaIds,
        Instant createdAt) {

    public enum Kind {
        VIDEO_ADAPTATION("video_adaptation"),
        ASSET_IMAGE("asset_image"),
        SCENE_IMAGE("scene_image"),
        /** 文章正文（任务书 #44 登记扩展：文本创作流接 lineage）。 */
        ARTICLE("article"),
        /** 朋友圈文案。 */
        MOMENTS_COPY("moments_copy"),
        /** 喜剧/脱口秀脚本。 */
        COMEDY_SCRIPT("comedy_script"),
        /** 智能创作助手问答引导（AI 中心 SSE 流）。 */
        ASSISTANT_GUIDE("assistant_guide");

        private final String db;

        Kind(String db) { this.db = db; }
        public String db() { return db; }

        public static Kind fromDb(String value) {
            for (Kind kind : values()) if (kind.db.equals(value)) return kind;
            throw new IllegalArgumentException("Unknown creation generation kind: " + value);
        }
    }

    public enum Mode {
        INDEPENDENT("independent"), TASK("task");
        private final String db;
        Mode(String db) { this.db = db; }
        public String db() { return db; }
        public static Mode fromDb(String value) {
            for (Mode mode : values()) if (mode.db.equals(value)) return mode;
            throw new IllegalArgumentException("Unknown creation generation mode: " + value);
        }
    }

    public enum Resolution {
        PLATFORM("platform"), BYOK("byok");
        private final String db;
        Resolution(String db) { this.db = db; }
        public String db() { return db; }
        public static Resolution fromDb(String value) {
            for (Resolution resolution : values()) if (resolution.db.equals(value)) return resolution;
            throw new IllegalArgumentException("Unknown creation generation resolution: " + value);
        }
    }
}
